common/include/pcl/impl/point_types.hpp

```diff
@@ -823,20 +825,28 @@ namespace pcl
   };
 
 
-
   PCL_EXPORTS std::ostream& operator << (std::ostream& os, const PointXY& p);
-  /** \brief A 2D point structure representing Euclidean xy coordinates.
+  /* \brief A 2D point structure representing Euclidean xy coordinates.
     * \ingroup common
     */
   struct PointXY
   {
-    float x = 0.f;
-    float y = 0.f;
-
+    union 
+    { 
+      float data[2]; 
+      struct 
+      { 
+        float x=0.0f; 
+        float y=0.0f; 
+      };
+    };
     inline PointXY() = default;
 
     inline PointXY(float _x, float _y): x(_x), y(_y) {}
-
+    
+    inline pcl::Vector2fMap getVector2fMap () { return (pcl::Vector2fMap (data)); }
+    inline pcl::Vector2fMapConst getVector2fMapConst () { return (pcl::Vector2fMapConst (data)); }
```

Comment:
Thoughts about adding these 2 lines to PCL_ADD_EIGEN_MAPS_POINT4D as well?

Reply:
But why so? I mean what will be the use?

Reply:
I'll add it though and see if it works.

Reply:
> But why so? I mean what will be the use?

Uniform API